api/envoy/extensions/filters/http/aws_request_signing/v3/aws_request_signing.proto

```diff
@@ -43,4 +43,9 @@ message AwsRequestSigning {
   // value set here would be used for signing whereas the value set in the HCM would be used
   // for host header forwarding which is not the desired outcome.
   string host_rewrite = 3;
+
+  // Instead of buffering the request to calculate the payload hash, use the literal string ``UNSIGNED-PAYLOAD``
+  // to calculate the payload hash. Not all services support this option. See the `S3
+  // <https://docs.aws.amazon.com/AmazonS3/latest/API/sig-v4-header-based-auth.html>`_ policy for details.
+  bool unsigned_payload = 4;
```

Comment:
Should we call this `use_unsigned_payload` or `skip_body_buffering`?

Reply:
I'll change it to `use_unsigned_payload`. Unsigned payload is a very specific signing algorithm variant that isn't widely supported. I would say that using unsigned payload implies buffering is not required, but there could be some future case that warrants disabling buffering without using the unsigned payload variant.

Reply:
I added two new counters that are only incremented when signing occurs in decodeData.